vraptor-core/src/main/java/br/com/caelum/vraptor/http/iogi/IogiParametersProvider.java

```diff
@@ -90,11 +90,12 @@ private Object instantiateOrAddError(Parameters parameters, List<Message> errors
 	private List<Target<Object>> createTargets(ControllerMethod method) {
 		Method javaMethod = method.getMethod();
 		List<Target<Object>> targets = new ArrayList<>();
+		TypeExtractor typeExtractor = new TypeExtractor(method.getController().getType());
```

Comment:
what about changing `TypeExtractor` to be a CDI managed bean?
so it could be specialized -- extensible